entrypoint.go

```diff
@@ -0,0 +1,426 @@
+
+	return err
+}
+
+func readArtifacts(fp string[]) ([]result.RunResult, error) {
+	file, err := os.ReadFile(fp)
+	if os.IsNotExist(err) {
+		return []result.RunResult{}, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	return []result.RunResult{{Key: fp, Value: string(file), ResultType: result.ArtifactsResultType}}, nil
+}
+
+
+// WritePostFile write the postfile
+func (e Entrypointer) WritePostFile(postFile string, err error) {
+	if err != nil && postFile != "" {
+		postFile += ".err"
+	}
+	if postFile != "" {
+		e.PostWriter.Write(postFile, "")
+	}
+	_, err = writeToTempFile(postFile)
+}
+
+// WriteExitCodeFile write the exitCodeFile
+func (e Entrypointer) WriteExitCodeFile(stepPath, content string) {
+	exitCodeFile := filepath.Join(stepPath, "exitCode")
+	e.PostWriter.Write(exitCodeFile, content)
+	_, err = writeToTempFile(exitCodeFile)
+
+}
+
+// waitingCancellation waiting cancellation file, if no error occurs, call cancelFunc to cancel the context
+func (e Entrypointer) waitingCancellation(ctx context.Context, cancel context.CancelFunc) error {
+	if err := e.Waiter.Wait(ctx, pod.DownwardMountCancelFile, true, false); err != nil {
+		return err
+	}
+	cancel()
+	return nil
+}
+
+
+// getStepResultPath gets the path to the step result
+func getStepResultPath(stepDir string, stepName string, resultName string) string {
+	return filepath.Join(stepDir, stepName, "results", resultName)
+}
+
+// outputRunResult returns the run reason for a termination
+func (e Entrypointer) outputRunResult(terminationReason string) result.RunResult {
+	return result.RunResult{
+		Key:        "Reason",
+		Value:      terminationReason,
+		ResultType: result.InternalTektonResultType,
+	}
+}
+
+// getStepArtifactsPath gets the path to the step artifacts
+func getStepArtifactsPath(stepDir string, containerName string) string {
+	return filepath.Join(stepDir, containerName, "artifacts", "provenance.json")
+}
+
+
+// getArtifactValues retrieves the values associated with a specified artifact reference.
+// It parses the provided artifact template, loads the corresponding step's artifacts, and extracts the relevant values.
+// If the artifact name is not specified in the template, the values of the first output are returned.
+func getArtifactValues(dir string, template string) (string, error) {
+	artifactTemplate, err := parseArtifactTemplate(template)
+
+	if err != nil {
+		return "", err
+	}
+
+	artifacts, err := loadStepArtifacts(dir, artifactTemplate.ContainerName)
+	if err != nil {
+		return "", err
+	}
+
+	// $(steps.stepName.outputs.artifactName) <- artifacts.Output[artifactName].Values
+	// $(steps.stepName.outputs) <- artifacts.Output[0].Values
+	var t []v1.Artifact
+	if artifactTemplate.Type == "outputs" {
+		t = artifacts.Outputs
+	} else {
+		t = artifacts.Inputs
+	}
+
+	if artifactTemplate.ArtifactName == "" {
+		marshal, err := json.Marshal(t[0].Values)
+		if err != nil {
+			return "", err
+		}
+		return string(marshal), err
+	}
+	for _, ar := range t {
```

Comment:
Consider refactoring the `getArtifactValues` function to use a map for storing artifacts, which would allow for more efficient retrieval of artifacts by name. [medium]